### PR TITLE
PMA 013-1 (board CRUD operations)

### PR DESCRIPTION
**Pull request by AlexanderSUS**
1. Add board CRUD operations

### CODE REVIEW

File: src/components/ModalConfirmButtons.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import { Button, Box } from '@mui/material';
+import { modalText } from '../constants/text';
+
+type Props = {
+  confirm: () => void;
+  close: () => void;
```
Comment (AnAtoliy-AA) [inline src/components/ModalConfirmButtons.tsx]: you can use voidFunction

File: src/components/forms/BoardForm.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
+import * as React from 'react';
+import Box from '@mui/material/Box';
+import { Button, TextField } from '@mui/material';
```
Comment (AnAtoliy-AA) [inline src/components/forms/BoardForm.tsx]: why you import Box at the upper line in other way

File: src/components/ModalConfirmButtons.tsx (hunk 1 above)
Comment (AlexanderSUS) [reply src/components/ModalConfirmButtons.tsx]: Thanks, didn't know about this option

File: src/components/forms/BoardForm.tsx (hunk 2 above)
Comment (AlexanderSUS) [reply src/components/forms/BoardForm.tsx]: It's autocomplete, will fix it in next commit

File: src/components/Board.tsx [hunk 3]
```diff
@@ -1,17 +1,41 @@
...
 
-const Board: React.FC<BoardProps> = ({ board: { id, title } }) => (
-  <Box key={id}>
-    <Box>
-      {title}
-    </Box>
-  </Box>
-);
+const Item = styled(Paper)(({ theme }) => ({
+  backgroundColor: theme.palette.mode === 'dark' ? '#1A2027' : '#fff',
```
Comment (allyksander) [inline src/components/Board.tsx:15]: I think, we can try to use data from muiTheme.ts here.

File: src/components/Header.tsx [hunk 4]
```diff
@@ -34,7 +34,7 @@ const Header: React.FC = () => {
             />
             <HeaderButton
               text={navText.newBoard}
-              onClick={() => { dispatch(openModal(NEW_BOARD)); }}
+              onClick={() => { dispatch(openModal({ content: NEW_BOARD, dataId: '', action: 'addBoard' })); }}
```
Comment (allyksander) [inline src/components/Header.tsx]: Is it only temporary text values?

**Review by allyksander — APPROVED** [review]

File: src/store/boardSlice.ts [hunk 5]
```diff
@@ -1,13 +1,11 @@
...
 import BoardService from '../api/boardServise';
-import {
-  Boards, VoidArgument, BoardState, BoardType,
-} from '../types/boards';
+import { Boards, VoidArgument, BoardState } from '../types/boards';
 import { ErrorResponseData, ValidationErrors } from '../types/response';
 import type { RootState } from './store';
 import initialState from '../constants/boards';
-import { ModalFormData } from '../types/modal';
+import { FormActionData } from '../types/modal';
```
Comment (Gibr) [inline src/store/boardSlice.ts:8]: change to "import type { FormActionData } ...." to avoid "Dependency cycle"

**Review by Gibr — APPROVED** [review]

File: src/components/Header.tsx (hunk 4 above)
Comment (AlexanderSUS) [reply src/components/Header.tsx]: I already fixed it in next branch

File: src/store/boardSlice.ts (hunk 5 above)
Comment (AlexanderSUS) [reply src/store/boardSlice.ts:8]: I fixed it in PMA-013-3

**Pull request merged**